ADAL/src/ADAuthenticationParameters+Internal.m

```diff
@@ -168,6 +168,150 @@ + (NSDictionary *)extractChallengeParameters:(NSString *)headerContents
+{
+    NSUInteger lastIndex = startIndex;
+    
+    unichar quotedChar = [headerContents characterAtIndex:startIndex];
+    if (quotedChar == '"' || quotedChar == '\'')
+    {
+        NSUInteger nextIndex = startIndex + 1;
+        while (nextIndex < headerContents.length) {
+            unichar c = [headerContents characterAtIndex:nextIndex];
+            
+            if (c == quotedChar) {
+                lastIndex = nextIndex;
+                break;
+            }
+            
+            nextIndex++;
+        }
+    }
+    
+    return lastIndex;
+}
+
++ (NSString *)extractBearerChallenge:(NSString *)headerContents
+{
+    NSRange range = [headerContents rangeOfString:@"Bearer "];
+    
+    if (range.location == NSNotFound)
+    {
+        return nil;
+    }
+    
+    NSUInteger nextIndex = range.location + range.length;
+    NSUInteger lastChallengeIndex = NSNotFound;
+    
+    BOOL possibleEndOfChallenge = NO;
+    BOOL nonSpaceOrEqualCharacterDetected = NO;
+    BOOL spaceCharacterDetected = NO;
+    
+    while (nextIndex < headerContents.length)
+    {
+        unichar c = [headerContents characterAtIndex:nextIndex];
+        
+        // Start of a quaoted string, lets get last index of it and continue iteration.
+        if (c == '"' || c == '\'')
+        {
+            nextIndex = [self quotedStringLastIndex:headerContents startIndex:nextIndex];
+            lastChallengeIndex = nextIndex;
+            nextIndex++;
+            continue;
+        }
+        
+        if (c == ',')
+        {
+            if (possibleEndOfChallenge)
+            {
+                // Next challenge found.
+                possibleEndOfChallenge = NO;
+                break;
+            }
+            
+            possibleEndOfChallenge = YES;
+            nonSpaceOrEqualCharacterDetected = NO;
+            spaceCharacterDetected = NO;
+        }
+        
+        if (!possibleEndOfChallenge)
+        {
+            lastChallengeIndex = nextIndex;
+        }
+        
+        if (c != ' ' && c != ',')
+        {
+            nonSpaceOrEqualCharacterDetected = YES;
+        }
+        
+        if (possibleEndOfChallenge && nonSpaceOrEqualCharacterDetected && c == ' ')
+        {
+            spaceCharacterDetected = YES;
+        }
+        
+        if (possibleEndOfChallenge && nonSpaceOrEqualCharacterDetected && c == '=')
+        {
+            // Next parameter found.
+            possibleEndOfChallenge = NO;
+            nonSpaceOrEqualCharacterDetected = NO;
+            spaceCharacterDetected = NO;
+            lastChallengeIndex = nextIndex;
+        }
+        
+        if (possibleEndOfChallenge && nonSpaceOrEqualCharacterDetected && spaceCharacterDetected && c != ',' && c != ' ' && c != ' ')
+        {
+            // Next challenge found.
+            possibleEndOfChallenge = NO;
+            break;
+        }
+        
+        nextIndex++;
+    }
+    
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the useless trailing whitespaces at the end of this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)